bpf/lib/proxy_hairpin.h

```diff
@@ -26,25 +26,25 @@ ctx_redirect_to_proxy_hairpin(struct __ctx_buff *ctx, __be16 proxy_port, const b
 	union macaddr host_mac = HOST_IFINDEX_MAC;
 	union macaddr router_mac = NODE_MAC;
 #endif
-	void *data_end = (void *)(long)ctx->data_end;
-	void *data = (void *)(long)ctx->data;
-	struct iphdr *ip4;
 	int ret = 0;
 
 	ctx_store_meta(ctx, CB_PROXY_MAGIC,
 		       MARK_MAGIC_TO_PROXY | (proxy_port << 16));
 	bpf_barrier(); /* verifier workaround */
 
-	if (!revalidate_data(ctx, &data, &data_end, &ip4))
-		return DROP_INVALID;
-
 	if (is_ipv6) {
 #ifdef ENABLE_IPV6
 		ret = ipv6_l3(ctx, ETH_HLEN, (__u8 *)&router_mac, (__u8 *)&host_mac,
 			      METRIC_EGRESS);
 #endif
 	} else {
 #ifdef ENABLE_IPV4
+		void *data, *data_end;
+		struct iphdr *ip4;
+
+		if (!revalidate_data(ctx, &data, &data_end, &ip4))
```

Comment:
and the `data*` variables are not used as well, so this can bi simplified even more :-)

Reply:
Ah no, that's part of our revalidation logic :). It's how the verifier understands that the L3 pointer is validated against the packet boundaries.

Reply:
Yes, sure. I just meant that you can allocate this inside the `revalidate_data` in this case, as `data*` is now used lower